Nautilus/Patchers/VehicleUpgradesPatcher.cs

```diff
@@ -239,7 +239,7 @@ private static void OnUpgradeModuleExoHull(Exosuit __instance, int slotID, TechT
             if (Exosuit.crushDepths.TryGetValue(techTypeInSlot, out depthToCheck) && depthToCheck > newCrushDepth)
             {
                 newCrushDepth = depthToCheck;
-                if (ExosuitUpgradeModules[techTypeInSlot].TryGetGadget(out UpgradeModuleGadget mdlGadget))
+                if (ExosuitUpgradeModules.TryGetValue(techTypeInSlot, out _) && ExosuitUpgradeModules[techTypeInSlot].TryGetGadget(out UpgradeModuleGadget mdlGadget))
```

Comment:
Why use a discard variable and not just use the output from the try get value? ``` 
if (ExosuitUpgradeModules.TryGetValue(techTypeInSlot, out upgrade) && upgrade.TryGetGadget(out UpgradeModuleGadget mdlGadget))
``` this way you aren't unnecessarily getting the value twice

Reply:
good question, lemme change that.

Reply:
Did this.